src/net.cpp

```diff
@@ -674,11 +676,19 @@ int V1TransportDeserializer::readHeader(const char *pch, unsigned int nBytes)
         hdrbuf >> hdr;
     }
     catch (const std::exception&) {
+        LogPrint(BCLog::NET, "HEADER ERROR - UNABLE TO DESERIALIZE, peer=%d\n", m_node_id);
```

Comment:
in commit 2ac16b110b

Adding logging is great, but I don't think this counts as a refactor. Also, the tests seem to require some changes, so maybe change the "without changing behavior" to something like "add logging" in the pull request title?

Reply:
I agree, this has grown and there is enough context in the PR description.  Changed.